### ISSUE TITLE
error CS0006: Metadata file '.../maccatalyst-arm64/StartupHookForFunctionalTest.dll' could not be found

### ISSUE THREAD

**Issue by EgorBo** [dotnet/runtime#80976]
`runtime-extra-platforms` fails on ios/tvos/maccatalyst with:
```
CSC : error CS0006: Metadata file '/Users/runner/work/1/s/artifacts/bin/StartupHookForFunctionalTest/Release/net8.0/iossimulator-x64/StartupHookForFunctionalTest.dll' could not be found [/Users/runner/work/1/s/src/tests/FunctionalTests/iOS/Simulator/StartupHook/iOS.Simulator.StartupHook.Test.csproj]
##[error]CSC(0,0): error CS0006: (NETCORE_ENGINEERING_TELEMETRY=Build) Metadata file '/Users/runner/work/1/s/artifacts/bin/StartupHookForFunctionalTest/Release/net8.0/iossimulator-x64/StartupHookForFunctionalTest.dll' could not be found

Build FAILED.
```

e.g. https://dev.azure.com/dnceng-public/public/_build/results?buildId=144305&view=logs&jobId=cbf7f906-06a1-5744-2b19-594b1486a781&j=07effcfd-c63d-5008-e811-03b29530f5cb&t=46984e67-96b8-5591-f4ad-dfc0b5406724

**Comment by mdh1418** [comment]
@steveisok @lambdageek 
I think this is because our iOS default workflow will remove the `$(OutDir)` 
> `Removing directory "/Users/mihw/runtime/runtime_ios/artifacts/bin/StartupHookForFunctionalTest/release/net8.0/iossimulator-arm64/".`

Afterwards, the `CoreCompile` target step sees `/Users/mihw/runtime/runtime_ios/artifacts/bin/StartupHookForFunctionalTest/release/net8.0/iossimulator-arm64/StartupHookForFunctionalTest.dll` as a reference but the directory was removed.

**Comment by steveisok** [comment]
@mdh1418 can we skip this test until there's a fix?

**Comment by mdh1418** [comment]
Yeah, I can skip it. I think the problem inherently because this test has a ProjectReference https://github.com/dotnet/runtime/blob/da03f608cbcc06d01694026ed8a3c882f9fce540/src/tests/FunctionalTests/iOS/Simulator/StartupHook/iOS.Simulator.StartupHook.Test.csproj#L19
but its not clear to me how this ends up with a reference to `/Users/mihw/runtime/runtime_ios/artifacts/bin/StartupHookForFunctionalTest/release/net8.0/iossimulator-arm64/StartupHookForFunctionalTest.dll` in `CoreCompile`d as opposed to the libraries tests which also have `ProjectReference` but dont have this problem.

**Comment by lambdageek** [comment]
ProjectReferences should work.
Why are we deleting that other directory?

**Comment by lambdageek** [comment]
Maybe the ProjectReference should be Private?
We want the hook dll to be copied into the functional test's bin directory

**Comment by mdh1418** [comment]
> ProjectReferences should work. Why are we deleting that other directory?

We were deleting that directory as a means to help conserve disk space, but upon some investigations in https://github.com/dotnet/runtime/pull/82313#issuecomment-1450874075, we still have plenty of disk space left even if we retain `$(OutDir)`.

**Comment by mdh1418** [comment]
@lambdageek @steveisok 
It doesn't look like making the `ProjectReference` private resolves the underlying problem.
From some investigation, the problem is that the Android build targets are being hit during the ResolveProjectRefernces target, probably because when resolving `StartupHookForFunctionalTest.csproj`, the `publish` target is hit, and because we use Publish as an `AfterTargets` https://github.com/dotnet/runtime/blob/f70f26f66cc30f46c255edc92e19d5b48d079a27/src/mono/msbuild/android/build/AndroidBuild.props#L9, the Android build is kicked off, thus hitting `_CopyTestArchive` and removing the output directory before we actually try to compile the actual project in `CoreCompile`
